src/fdb5/tools/fdb-list.cc

```diff
@@ -8,106 +8,110 @@
  * does it submit to any jurisdiction.
  */
 
+#include <map>
+#include <memory>
+#include <string>
 #include <unordered_set>
+#include <utility>
+#include <vector>
 
+#include "eckit/exception/Exceptions.h"
+#include "eckit/log/JSON.h"
+#include "eckit/log/Log.h"
 #include "eckit/option/CmdArgs.h"
 #include "eckit/option/SimpleOption.h"
-#include "eckit/option/CmdArgs.h"
-#include "eckit/log/JSON.h"
 
+#include "fdb5/LibFdb5.h"
 #include "metkit/hypercube/HyperCube.h"
+#include "metkit/mars/MarsRequest.h"
 
 #include "fdb5/api/FDB.h"
 #include "fdb5/api/helpers/FDBToolRequest.h"
+#include "fdb5/api/helpers/ListElement.h"
+#include "fdb5/api/helpers/ListIterator.h"
 #include "fdb5/database/Index.h"
 #include "fdb5/rules/Schema.h"
 #include "fdb5/tools/FDBVisitTool.h"
 
 using namespace eckit;
 using namespace eckit::option;
 
-namespace fdb5 {
-namespace tools {
+namespace fdb5::tools {
 
 //----------------------------------------------------------------------------------------------------------------------
 
-class FDBList : public FDBVisitTool {
-
-  public: // methods
-
-    FDBList(int argc, char **argv) :
-        FDBVisitTool(argc, argv, "class,expver"),
-        location_(false),
-        timestamp_(false),
-        length_(false),
-        full_(false),
-        porcelain_(false),
-        json_(false) {
-
+class FDBList: public FDBVisitTool {
+public:  // methods
+    FDBList(int argc, char** argv): FDBVisitTool(argc, argv, "class,expver") {
         options_.push_back(new SimpleOption<bool>("location", "Also print the location of each field"));
         options_.push_back(new SimpleOption<bool>("timestamp", "Also print the timestamp when the field was indexed"));
         options_.push_back(new SimpleOption<bool>("length", "Also print the field size"));
         options_.push_back(new SimpleOption<bool>("full", "Include all entries (including masked duplicates)"));
-        options_.push_back(new SimpleOption<bool>("porcelain", "Streamlined and stable output for input into other tools"));
+        options_.push_back(
+            new SimpleOption<bool>("porcelain",
+                                   "Streamlined and stable output. Useful as input for other tools or scripts."
+                                   "Incompatible with options: location, timestamp, and length"));
         options_.push_back(new SimpleOption<bool>("json", "Output available fields in JSON form"));
```

Comment:
Since you already updated the doc: This also needs to mention the interaction of `--json` with `--porcelain` and the resulting incompatibilities